test/test_compat.py

```diff
@@ -53,6 +55,27 @@ def test_all_present(self):
             dir(youtube_dlc.compat))) - set(['unicode_literals'])
         self.assertEqual(all_names, sorted(present_names))
 
+    def test_compat_urllib_parse_quote(self):
+        self.assertEqual(compat_urllib_parse_quote('abc def'), 'abc%20def')
+        self.assertEqual(compat_urllib_parse_quote('/~user/abc+def'), '/%7Euser/abc%2Bdef')
+        self.assertEqual(compat_urllib_parse_quote('/~user/abc+def', safe='/~+'), '/~user/abc+def')
```

Comment:
(See explaining comment.)

```suggestion
        self.assertEqual(compat_urllib_parse_quote('/abc+def'), '/abc%2Bdef')
        self.assertEqual(compat_urllib_parse_quote('/abc+def', safe='+'), '%2Fabc+def')
```